src/filetreemodel.cpp

```diff
@@ -969,8 +970,12 @@ void FileTreeModel::updateFileItem(FileTreeItem& item, const MOShared::FileEntry
 
 bool FileTreeModel::shouldShowFile(const FileEntry& file) const
 {
-  if (showConflictsOnly() && (file.getAlternatives().size() == 0)) {
-    // only conflicts should be shown, but this file is not conflicted
+  if (showConflictsOnly() &&
+      ((file.getAlternatives().size() == 0) ||
+       fs::path(file.getName())
+               .extension()
+               .compare(ModInfo::s_HiddenExt.toStdWString()) == 0)) {
+    // only conflicts should be shown, but this file is hidden or not conflicted
```

Comment:
This should be a hotpath, I don't know how fast the `fs:path` conversion to compare the extension is. Might be worth profiling this under a large setup. I know we got massive wins by using a faster string comparison in the past

Reply:
Do you think `file.getName().ends_with(ModInfo::s_HiddenExt.toStdWString())` would suffice? That's what I used [here](https://github.com/ModOrganizer2/modorganizer/pull/2146/files). I only used the `fs::path` conversion here to be consistent with similar existing code that I used as a reference.

Reply:
The check needs to be case insensitive, which is what can make it costly. If it isn't a problem on a larger setup we don't need to optimize it. Just need to check it isn't a problem.

Reply:
The extension being anything other than fully lowercase would be an unusual edge case, but I suppose we should still account for that possibility. I'm not sure how to go about profiling it though. If we already know it's a hot path, I'd say it's worth optimizing regardless. How about `boost::algorithm::iends_with(file.getName(), ModInfo::s_HiddenExt.toStdWString()))`? It's case insensitive and doesn't require a `fs::path` conversion.

Reply:
```cpp
QString::fromStdWString(file.getName()).endsWith(ModInfo::s_HiddenExt, Qt::CaseInsensitive))
```
Not sure why `FileEntry` uses `std::w/string` types. Also the less boost usage, the better.